data/tribes/buildings/productionsites/amazons/ferry_yard/init.lua

```diff
@@ -44,10 +44,10 @@ wl.Descriptions():new_productionsite_type {
          -- TRANSLATORS: Completed/Skipped/Did not start working because ...
          descname = _("working"),
          actions = {
+            "return=skipped unless fleet needs ferry",
             "sleep=duration:20s",
-            "callworker=buildferry_1",
             "consume=log:3",
-            "callworker=buildferry_2"
+            "callworker=buildferry"
```

Comment:
Does this mean that we may now lose the materials if no suitable place is found to build a ferry? (OK, old check had a time-of-check-time-of-use race too, but it was still better, wasn't it?)

Reply:
The `return=skipped unless fleet needs ferry` conditional already checks that there is a ferry fleet interface nearby that actually needs a ferry and is not blocked by immovables etc. So if this check passes then we know that the worker will really be able to build a ferry.